### ISSUE TITLE
Massive performance loss after restoring checkpoint

### ISSUE THREAD

**Issue by hmenke** [espressomd/espresso#1969]
```python
from __future__ import print_function
import numpy as np
import espressomd
import espressomd.checkpointing
import time

checkpoint = espressomd.checkpointing.Checkpointing(checkpoint_id="checkpoint")

start_sample = 0
checkpoint.register("start_sample")

system = espressomd.System(box_l = [10,10,10])
system.periodicity = [1,1,1]
system.cell_system.skin = 0.1
system.time_step = 0.01
checkpoint.register("system")

if not checkpoint.has_checkpoints():
    for i in range(10):
        system.part.add(pos=system.box_l * np.random.random(3))
    checkpoint.register("system.part")

if not checkpoint.has_checkpoints():
    checkpoint.save(checkpoint_index=0)

restarted = False
if checkpoint.has_checkpoints():
    print("Restoring Equilibration Checkpoint")
    checkpoint.load(checkpoint_index=0)
    print("Checkpoint loaded")
    restarted = True

for k in range(start_sample,200):
    start = time.time()
    system.integrator.run(steps=5000, reuse_forces=restarted)
    print("Step: {}, Time: {}".format(k,time.time() - start))
    restarted = False
    start_sample = k + 1
    checkpoint.save(checkpoint_index=0)
```
After restoring the checkpoint, integration is a **factor of 5 to 6 slower!**

This is reproducible in the latest [`espressomd/espresso-ubuntu`](https://hub.docker.com/r/espressomd/espresso-ubuntu/) container with `maxset.hpp`.
```
root@9d314b7fbda0:/tmp/espresso/build# git show
commit bbf1a43c6db393fa473a5824385aa87b3563a2a1
Merge: e9b7a5c 069db89
Author: Florian Weik <florianweik@gmail.com>
Date:   Thu May 3 21:38:12 2018 +0200

    Merge pull request #1967 from RudolfWeeber/id
    
    py: Make ParticleHandle.id read-only

root@9d314b7fbda0:/tmp/espresso/build# ./pypresso test.py 
Step: 0, Time: 0.0387508869171
Step: 1, Time: 0.0388331413269
Step: 2, Time: 0.0390479564667
Step: 3, Time: 0.0391790866852
Step: 4, Time: 0.0393280982971
Step: 5, Time: 0.0391201972961
Step: 6, Time: 0.0390129089355
Step: 7, Time: 0.0390889644623
^CTraceback (most recent call last):
  File "test.py", line 35, in <module>
    system.integrator.run(steps=5000, reuse_forces=restarted)
KeyboardInterrupt
root@9d314b7fbda0:/tmp/espresso/build# ./pypresso test.py 
Restoring Equilibration Checkpoint
Checkpoint loaded
Step: 8, Time: 0.229771852493
Step: 9, Time: 0.269323825836
Step: 10, Time: 0.227708816528
Step: 11, Time: 0.228987932205
Step: 12, Time: 0.232725858688
Step: 13, Time: 0.226660966873
Step: 14, Time: 0.22985291481
Step: 15, Time: 0.229024887085
Step: 16, Time: 0.229374885559
^CTraceback (most recent call last):
  File "test.py", line 35, in <module>
    system.integrator.run(steps=5000, reuse_forces=restarted)
KeyboardInterrupt
root@9d314b7fbda0:/tmp/espresso/build# 
```

**Comment by fweik** [comment]
At least part of the issue is that verlet lists are disabled after loading the checkpoint.

**Comment by KaiSzuttor** [comment]
Fixed in #1974
